src/main/kotlin/org/web3j/strategy/CustomWaitingStrategy.kt

```diff
@@ -0,0 +1,285 @@
+     *
+     * @return this
+     */
+    fun usingTls(): CustomWaitingStrategy {
+        tlsEnabled = true
+        return this
+    }
+
+    /**
+     * Authenticate with HTTP Basic Authorization credentials.
+     *
+     * @param username the username
+     * @param password the password
+     * @return this
+     */
+    fun withBasicCredentials(username: String?, password: String?): CustomWaitingStrategy {
+        this.username = username
+        this.password = password
+        return this
+    }
+
+    /**
+     * Set the HTTP connections read timeout.
+     *
+     * @param timeout the timeout (minimum 1 millisecond)
+     * @return this
+     */
+    fun withReadTimeout(timeout: Duration): CustomWaitingStrategy {
+        require(timeout.toMillis() >= 1) { "you cannot specify a value smaller than 1 ms" }
+        readTimeout = timeout
+        return this
+    }
+
+    /**
+     * Waits for the response to pass the given predicate
+     * @param responsePredicate The predicate to test the response against
+     * @return this
+     */
+    fun forResponsePredicate(responsePredicate: Predicate<String>?): CustomWaitingStrategy {
+        this.responsePredicate = responsePredicate
+        return this
+    }
+
+    override fun waitUntilReady() {
+        val containerName = waitStrategyTarget.containerInfo.name
+        val livenessCheckPort = livenessPort.map { originalPort: Int? ->
+            waitStrategyTarget.getMappedPort(
+                originalPort!!
+            )
+        }.orElseGet {
+            val livenessCheckPorts = livenessCheckPorts
+            if (livenessCheckPorts == null || livenessCheckPorts.isEmpty()) {
+                // log.warn("{}: No exposed ports or mapped ports - cannot wait for status", containerName)
+                return@orElseGet -1
+            }
+            livenessCheckPorts.iterator().next()
+        }
+        if (null == livenessCheckPort || -1 == livenessCheckPort) {
+            return
+        }
+        val uri = buildLivenessUri(livenessCheckPort).toString()
+        // log.info("{}: Waiting for {} seconds for URL: {}", containerName, startupTimeout.seconds, uri)
+
+        // try to connect to the URL
+        try {
+            Unreliables.retryUntilSuccess(
+                startupTimeout.seconds.toInt(), TimeUnit.SECONDS
+            ) {
+                rateLimiter.doWhenReady {
+                    try {
+                        val connection =
+                            URL(uri).openConnection() as HttpURLConnection
+                        connection.readTimeout = Math.toIntExact(readTimeout.toMillis())
+
+                        // authenticate
+                        if (!Strings.isNullOrEmpty(username)) {
+                            connection.setRequestProperty(HEADER_AUTHORIZATION, buildAuthString(username, password))
+                            connection.useCaches = false
+                        }
+                        connection.requestMethod = "OPTIONS"
```

Comment:
This was not the idea, we cannot reuse this if you hardcode the HTTP method.

The thing was to add a `withMethod(String)` where con can specify it.